platform/viewer/src/routes/Mode/Mode.tsx

```diff
@@ -42,45 +49,42 @@ function defaultRouteInit({ servicesManager, studyInstanceUIDs, dataSource }) {
 
   unsubscriptions.push(instanceAddedUnsubscribe);
 
-  const { unsubscribe: seriesAddedUnsubscribe } = DisplaySetService.subscribe(
-    DisplaySetService.EVENTS.DISPLAY_SETS_CHANGED,
-    displaySets => {
-      if (!displaySets || !displaySets.length) return;
-      const studyMap = {};
-      // Prior studies don't quite work properly yet, but the studies list
-      // is at least being generated and passed in.
-      const studies = displaySets.reduce((prev, curr) => {
-        const { StudyInstanceUID } = curr;
-        if (!studyMap[StudyInstanceUID]) {
-          const study = DicomMetadataStore.getStudy(StudyInstanceUID);
-          studyMap[StudyInstanceUID] = study;
-          prev.push(study);
-        }
-        return prev;
-      }, []);
-      // The assumption is that the display set at position 0 is the first
-      // study being displayed, and is thus the "active" study.
-      const activeStudy = studies[0];
-      HangingProtocolService.run({ studies, activeStudy, displaySets });
-      // Don't fire off any more hanging protocol service initiates since
-      // it may well re-layout the study.
-      seriesAddedUnsubscribe();
-    }
+  const allRetrieves = studyInstanceUIDs.map(StudyInstanceUID =>
+    dataSource.retrieve.series.metadata({ StudyInstanceUID })
   );
-  // Add the unsubscription to the list in case the cancel happens before the
-  // service is done.
-  unsubscriptions.push(seriesAddedUnsubscribe);
 
   // The hanging protocol matching service is fairly expensive to run multiple
   // times, and doesn't allow partial matches to be made (it will simply fail
-  // to display anything if a required match failes), so hold off the matches
+  // to display anything if a required match fails), so hold off the matches
   // here until the entire study is ready.
-  DisplaySetService.holdChangeEvents();
-  const allRetrieves = studyInstanceUIDs.map(StudyInstanceUID =>
-    dataSource.retrieve.series.metadata({ StudyInstanceUID })
-  );
+
   Promise.allSettled(allRetrieves).then(() => {
-    DisplaySetService.fireHoldChangeEvents();
+    const displaySets = DisplaySetService.getActiveDisplaySets();
```

Comment:
You have re-introduced a bug here.  The retrieval of instances does NOT guarantee that display sets are generated yet, so you can randomly get matching or non-matching display sets.  For example, suppose we have
allRetrieves = [series1, series2]
Now, series2 has delivered it's display set in the getActiveDisplaySets(), but series1 has it's display set delivery still on the event queue.  So, displaySets = [series2 display set], which will, even on a 1x1 layout with no other constraints end up showing series 2 instead of series 1 on initial display.  I can show you more complex cases which completely fail to run, including your pt display set.  USUALLY it works on a fast retrieve, because the ordering of the event queue is such that it is usually done in time here, but randomly it will fail.